fvm/transactionInvocator.go

```diff
@@ -196,7 +204,44 @@ func (i *TransactionInvocator) Process(
 	return nil
 }
 
-func (i *TransactionInvocator) valueDeclarations(ctx *Context, env *hostEnv) []runtime.ValueDeclaration {
+func (i *TransactionInvocator) deductTransactionFees(env *hostEnv, proc *TransactionProcedure) error {
+	if !env.ctx.TransactionFeesEnabled {
+		return nil
+	}
+
+	invocator := NewTransactionContractFunctionInvocator(
+		common.AddressLocation{
+			Address: common.BytesToAddress(env.ctx.Chain.ServiceAddress().Bytes()),
+			Name:    flowServiceAccountContract,
+		},
+		deductFeesContractFunction,
+		[]interpreter.Value{
+			interpreter.NewAddressValue(common.BytesToAddress(proc.Transaction.Payer.Bytes())),
+		},
+		[]sema.Type{
+			sema.AuthAccountType,
+		},
+		zerolog.Nop(),
+	)
+	_, err := invocator.Invoke(env, proc)
+
+	if err != nil {
+		// TODO: Fee value is currently a constant. this should be changed when it is not
+		return errors.NewTransactionFeeDeductionFailedError(proc.Transaction.Payer, DefaultTransactionFees.ToGoValue().(uint64), err)
```

Comment:
```
DefaultTransactionFees.ToGoValue().(uint64)
```

this is prone to accidental changes, I think we should wrap the type casting with ok argument and gracefully return with out deducting fees.